packages/vite/src/node/utils.ts

```diff
@@ -1045,19 +1049,27 @@ export function resolveServerUrls(
       })
   }
 
-  const cert =
-    httpsOptions?.cert && !Array.isArray(httpsOptions.cert)
-      ? new crypto.X509Certificate(httpsOptions.cert)
-      : undefined
-  const hostnameFromCert = cert?.subjectAltName
-    ? extractHostnamesFromSubjectAltName(cert.subjectAltName)
-    : []
+  const certs = httpsOptions?.cert ? arraify(httpsOptions.cert) : []
+  const hostnameFromCert = certs
+    .map((cert) => {
+      try {
+        const buf = bufferify(cert)
+        return new crypto.X509Certificate(buf)
+      } catch {
+        return null
+      }
+    })
+    .flatMap((cert) =>
+      cert && cert.subjectAltName
+        ? extractHostnamesFromSubjectAltName(cert.subjectAltName)
+        : [],
+    )
```

Comment:
Would you extract this part as a function so that we don't have to use mocks in the tests?